File: BusBud/BusBud/DataModels/Departure.swift
//
//  Departure.swift
//  BusBud
//
//  Created by Ali Pourhadi on 2020-10-08.
//

import Foundation

struct Departure: Codable {
    let id: String
    let sourceID: Int
    let operatorID: String
    let originLocationID: Int
    let destinationLocationID: Int
    let departureClass: String
    let className: String
    let availableSeats: Int
    let prices: Prices
    let ticketTypes: [String]
    let departureTimezone: String
    let arrivalTimezone: String
    let departureTime: String
    let arrivalTime: String

    enum CodingKeys: String, CodingKey {
        case id, prices
        case sourceID = "source_id"
        case operatorID = "operator_id"
        case originLocationID = "origin_location_id"
        case destinationLocationID = "destination_location_id"
        case departureClass = "class"
        case className = "class_name"
        case availableSeats = "available_seats"
        case ticketTypes = "ticket_types"
        case departureTimezone = "departure_timezone"
        case arrivalTimezone = "arrival_timezone"
        case departureTime = "departure_time"
        case arrivalTime = "arrival_time"
    }
}

File: app/src/main/java/com/ruimendes/busbud/model/Departure.kt
package com.ruimendes.busbud.model

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Departure(
    @Json(name = "id")
    val id: String,
    @Json(name = "class_name")
    val className: String,
    @Json(name = "available_seats")
    val availableSeats: Int,
    @Json(name = "prices")
    val prices: Price,
    @Json(name = "departure_time")
    val departureTime: String,
    @Json(name = "arrival_time")
    val arrivalTime: String,
    @Json(name = "operator_id")
    val operatorId: String,
    @Json(name = "origin_location_id")
    val originLocationId: String,
    @Json(name = "destination_location_id")
    val destinationLocationId: String,
): Parcelable